src/main/cpp/src/api/genomicsdb_json_processor.cc

```diff
@@ -0,0 +1,196 @@
+    json_doc->SetObject();
+    auto& allocator = json_doc->GetAllocator();
+    for (auto const& [sample_name, fields] : m_sample_info) {
+      auto i = 0u;
+      rapidjson::Value sample_value = rapidjson::Value(rapidjson::kObjectType);
+      rapidjson::Value sample_name_value(rapidjson::StringRef(sample_name.c_str()), allocator);
+      if (m_payload_mode == samples_with_ncalls) {
+        auto chrom_val = reinterpret_cast<rapidjson::Value *>(fields[0]);
+        json_doc->AddMember(sample_name_value, chrom_val->Size(), allocator);
+      } else {
+        rapidjson::Value field_values = rapidjson::Value(rapidjson::kObjectType);
+        field_values.AddMember("CHROM",
+                               *(reinterpret_cast<rapidjson::Value *>(fields[i++])),
+                               allocator);
+        field_values.AddMember("POS",
+                               *(reinterpret_cast<rapidjson::Value *>(fields[i++])),
+                               allocator);
+      
+        for (auto field_name : m_field_names) {
+          rapidjson::Value field_name_value(rapidjson::StringRef(field_name.c_str()), allocator);
+          field_values.AddMember(field_name_value,
+                                 *(reinterpret_cast<rapidjson::Value *>(fields[i++])),
+                                 allocator);
+        }
+        json_doc->AddMember(sample_name_value, field_values, allocator);
+      }
+    }
+  }
+
+  rapidjson::StringBuffer *buffer = reinterpret_cast<rapidjson::StringBuffer *>(m_json_buffer);
+  buffer->Clear();
+  rapidjson::Writer<rapidjson::StringBuffer> writer(*buffer);
+  json_doc->Accept(writer);
+  
+  return buffer->GetString();
+}
+
+void JSONVariantCallProcessor::process(const interval_t& interval) {
+  if (!m_is_initialized) {
+    m_is_initialized = true;
+    auto& genomic_field_types = get_genomic_field_types();
+    for (auto& field_type_pair : *genomic_field_types) {
+      std::string field_name = field_type_pair.first;
+      genomic_field_type_t field_type = field_type_pair.second;
+      if (field_name.compare("END") && field_name.compare("REF") && field_name.compare("ALT")) {
+        m_field_names.push_back(field_name);
+      }
+    }
+  }
+}
+
+void JSONVariantCallProcessor::process(const std::string& sample_name,
+                                       const int64_t* coordinates,
+                                       const genomic_interval_t& genomic_interval,
+                                       const std::vector<genomic_field_t>& genomic_fields) {
+  m_num_calls++;
+  if (m_payload_mode == just_ncalls) return;
+
+  rapidjson::Document *json_doc = TO_JSON_DOCUMENT(m_json_document);
+  auto& allocator = json_doc->GetAllocator();
+  auto sample_info = m_sample_info.find(sample_name);
+  if (sample_info == m_sample_info.end()) {
+    // Create Value Arrays and add to map
+    std::vector<void *> fields;
+    rapidjson::Value *chrom = new rapidjson::Value(rapidjson::kArrayType);
+    fields.push_back(chrom);
```

Comment:
Question about this -- for sample with ncalls would it be more efficient to increment a counter instead of storing each of the chrom/contig values?